runtime/worker.rs

```diff
@@ -261,8 +261,27 @@ impl MainWorker {
     cx: &mut Context,
   ) -> Poll<Result<(), AnyError>> {
     // We always poll the inspector if it exists.
-    let _ = self.inspector.as_mut().map(|i| i.poll_unpin(cx));
```

Comment:
The logic looks correct to me. It could be a bit easier to read, it seems that it boils down to:

```
if js_runtime_is_alive {
  if has_inspector { let _ = inspector.poll(); }
  js_runtime.poll()
} else if has_inspector && inspector.has_active_sessions {
  inspector.poll() 
}
```

I guess you wanted to not duplicate the `inspector.poll()` call, which I can understand.

Reply:
Yeah, precisely. We also didn't have a helper method to tell if the runtime is "alive" or not it's precisely the output of poling the runtime that provided that info, we could promote this code in `JsRuntime::poll_event_loop()` to a helper method:
```rust
    let state = state_rc.borrow();
    let has_pending_ops = !state.pending_ops.is_empty();

    let has_pending_dyn_imports = !{
      state.preparing_dyn_imports.is_empty()
        && state.pending_dyn_imports.is_empty()
    };
    let has_pending_dyn_module_evaluation =
      !state.pending_dyn_mod_evaluate.is_empty();
    let has_pending_module_evaluation = state.pending_mod_evaluate.is_some();

    if !has_pending_ops
      && !has_pending_dyn_imports
      && !has_pending_dyn_module_evaluation
      && !has_pending_module_evaluation
    {
      return Poll::Ready(Ok(()));
    }
 ```